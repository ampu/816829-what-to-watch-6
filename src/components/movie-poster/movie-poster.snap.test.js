import React from 'react';

import {render} from '@testing-library/react';

import MoviePoster from './movie-poster';
import PosterSize from '../../constants/poster-size';

describe(`MoviePoster component`, () => {
  it(`Should render according to the snapshot with no movie poster provided`, () => {
    const {container} = render(<MoviePoster/>);
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot with movie poster provided`, () => {
    const movie = {
      title: `fakeTitle`,
      poster: `fakePoster`,
    };
    const {container} = render(<MoviePoster movie={movie}/>);
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot with movie poster and small poster size provided`, () => {
    const movie = {
      title: `fakeTitle`,
      poster: `fakePoster`,
    };
    const {container} = render(<MoviePoster movie={movie} size={PosterSize.SMALL}/>);
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot with movie poster and big poster size provided`, () => {
    const movie = {
      title: `fakeTitle`,
      poster: `fakePoster`,
    };
    const {container} = render(<MoviePoster movie={movie} size={PosterSize.BIG}/>);
    expect(container).toMatchSnapshot();
  });
});
