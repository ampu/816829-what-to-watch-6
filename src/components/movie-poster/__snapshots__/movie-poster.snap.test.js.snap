// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster component Should render according to the snapshot with movie poster and big poster size provided 1`] = `
<div>
  <div
    class="movie-card__poster movie-card__poster--big"
  >
    <img
      alt="fakeTitle poster"
      height="327"
      src="fakePoster"
      width="218"
    />
  </div>
</div>
`;

exports[`MoviePoster component Should render according to the snapshot with movie poster and small poster size provided 1`] = `
<div>
  <div
    class="movie-card__poster movie-card__poster--small"
  >
    <img
      alt="fakeTitle poster"
      height="327"
      src="fakePoster"
      width="218"
    />
  </div>
</div>
`;

exports[`MoviePoster component Should render according to the snapshot with movie poster provided 1`] = `
<div>
  <div
    class="movie-card__poster"
  >
    <img
      alt="fakeTitle poster"
      height="327"
      src="fakePoster"
      width="218"
    />
  </div>
</div>
`;

exports[`MoviePoster component Should render according to the snapshot with no movie poster provided 1`] = `
<div>
  
</div>
`;
